src/Objects/SimpleGoogleMaps.php

```diff
@@ -1,153 +1,236 @@
+
+        if (($results = $this->cache->get($cacheKey)) === false) {
+            $response = (new Client())->request('GET', $queryUrl);
+            $results = json_decode($response->getBody());
+        }
+
+        if (!$results) {
+            throw new \Exception('Unable to parse response.');
+        }
+
+        if (!empty($results->error_message)) {
+            throw new \Exception('Error from Google Maps API: '.$results->error_message);
+        }
+
+        if (!$results->results) {
+            return;
+        }
+
+        $result = $results->results[0];
+
+        if (!$this->allowPartialMatches) {
+            if (isset($result->partial_match) && $result->partial_match) {
+                return;
+            }
+        }
+
+        $this->cache->set($cacheKey, $results);
+
+        return new LatLong($result->geometry->location->lat, $result->geometry->location->lng);
+    }
+
+    /**
+     * Look ups an LatLng location, and returns a string containing the address of that location.
+     *
+     * @param LatLong $latLong
+     *
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     *
+     * @return string
+     */
+    public function reverseGeocode(LatLong $latLong)
+    {
+        $queryUrl = $this->authObject->applyToUrl(
+            $this->baseUrl.'geocode/json?latlng='.urlencode($latLong->lat.','.$latLong->long)
+        );
+
+        $cacheKey = sha1(serialize([__FUNCTION__, func_get_args()]));
+
+        if (($results = $this->cache->get($cacheKey)) === false) {
+            $response = (new Client())->request('GET', $queryUrl);
+            $results = json_decode($response->getBody());
+        }
+
+        if (!$results) {
+            throw new \Exception('Unable to parse response.');
+        }
+
+        if (!empty($results->error_message)) {
+            throw new \Exception('Error from Google Maps API: '.$results->error_message);
+        }
+
+        if (!$results->results) {
+            return;
+        }
+
+        $result = $results->results[0];
+
+        if (!isset($result->formatted_address)) {
+            return;
+        }
+
+        $this->cache->set($cacheKey, $results);
+
+        return (string) $result->formatted_address;
+    }
+
+    /**
+     * Retrieves directions between two points ($from, and $to), using the travel mode
+     * defined by the $travelMode (TravelMode enum).
+     *
+     * @param LatLong|string $from
+     * @param LatLong|string $to
+     * @param string         $travelMode
+     *
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     *
+     * @return Journey|null
+     */
+    public function directions($from, $to, $travelMode = TravelMode::DRIVING)
+    {
+        if (is_object($from) && get_class($from) === LatLong::class) {
+            $from = $from->lat.','.$from->long;
+        }
+
+        if (is_object($to) && get_class($to) === LatLong::class) {
+            $to = $to->lat.','.$to->long;
+        }
```

Comment:
Nothing breaking at all, but is it worth letting the LatLong object determine what it's own string representation looks like? Comma separated LatLong pairs are fairly commonplace and bound to be used elsewhere.

Just out of curiosity, is there any intentional benefit of performing `is_object($from) && get_class($from) === LatLong::class` over `$from instanceof LatLong`?